lib/src/inbox_message_view.dart

```diff
@@ -6,43 +6,51 @@ import 'package:flutter/widgets.dart';
 import 'airship.dart';
 
 class InboxMessageView extends StatefulWidget {
-  final InboxMessageViewCreatedCallback onViewCreated;
+  final InboxMessage message;
+  final onLoadStarted;
+  final onLoadFinished;
+  final onLoadError;
+  final onClose;
+
 
   InboxMessageView({
     Key key,
-    @required this.onViewCreated
+    @required this.message,
+    this.onLoadStarted,
+    this.onLoadFinished,
+    this.onLoadError,
+    this.onClose,
   });
 
   @override
   _InboxMessageViewState createState() => _InboxMessageViewState();
 }
 
-typedef void InboxMessageViewCreatedCallback(InboxMessageViewController controller);
-
-class InboxMessageViewController {
-
+class _InboxMessageViewState extends State<InboxMessageView> {
   MethodChannel _channel;
 
-  InboxMessageViewController.init(int id) {
+  Future<void> onPlatformViewCreated(id) async {
     _channel =  new MethodChannel('com.airship.flutter/InboxMessageView_$id');
+    if (widget.onLoadStarted != null) {
+      widget.onLoadStarted();
+    }
+    loadMessage(widget.message).then((message) {
+      if (widget.onLoadFinished != null) {
+        widget.onLoadFinished();
+      }
+    });
   }
 
   Future<void> loadMessage(InboxMessage message) async {
     if (message == null) {
+      if (widget.onLoadError != null) {
+        widget.onLoadError(ArgumentError.notNull('message'));
```

Comment:
Look at the react implementation. Should be from the native layer if it fails to load the message.